src/algebra/homology/local_cohomology.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2023 Emily Witt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Emily Witt, Scott Morrison
+-/
+import ring_theory.ideal.basic
+import algebra.category.Module.colimits
+import algebra.category.Module.projective
+import category_theory.abelian.ext
+
+/-!
+# Local cohomology.
+
+We just give one definition, deferring for later the other equivalent definitions,
+and basic properties.
+-/
+
+open opposite
+open category_theory
+open category_theory.limits
+
+noncomputable theory
+
+variables (R : Type) [comm_ring R] (I : ideal R) (M : Module.{0} R)
+
+local attribute [ext] quiver.hom.unop_inj
+
+/-- The functor from the natural numbers that gives the R-modules R/I^t -/
+def ring_mod_powers : ℕ ⥤ (Module.{0} R)ᵒᵖ :=
```

Comment:
IF you like the universe change, all these `Module.{0}`s should be `Module.{u}`s.